crates/db_views/src/post_view.rs

```diff
@@ -626,66 +647,238 @@ mod tests {
+  }
+
+  #[test]
+  #[serial]
+  fn post_listing_like() {
+    let conn = establish_unpooled_connection();
+    let data = init_data(&conn);
+
+    let post_like_form = PostLikeForm {
+      post_id: data.inserted_post.id,
+      person_id: data.inserted_person.id,
+      score: 1,
+    };
+
+    let inserted_post_like = PostLike::like(&conn, &post_like_form).unwrap();
+
+    let expected_post_like = PostLike {
+      id: inserted_post_like.id,
+      post_id: data.inserted_post.id,
+      person_id: data.inserted_person.id,
+      published: inserted_post_like.published,
+      score: 1,
+    };
     assert_eq!(expected_post_like, inserted_post_like);
+
+    let like_removed =
+      PostLike::remove(&conn, data.inserted_person.id, data.inserted_post.id).unwrap();
     assert_eq!(1, like_removed);
-    assert_eq!(1, num_deleted);
+    cleanup(data, &conn);
+  }
+
+  #[test]
+  #[serial]
+  fn post_listing_person_language() {
+    let conn = establish_unpooled_connection();
+    let data = init_data(&conn);
+
+    let spanish_id = Language::read_id_from_code(&conn, "es").unwrap();
+    let post_spanish = PostForm {
+      name: "asffgdsc".to_string(),
+      creator_id: data.inserted_person.id,
+      community_id: data.inserted_community.id,
+      language_id: Some(spanish_id),
+      ..PostForm::default()
+    };
+
+    Post::create(&conn, &post_spanish).unwrap();
+
+    let my_person_form = PersonForm {
+      name: "Reverie Toiba".to_string(),
+      public_key: Some("pubkey".to_string()),
+      ..PersonForm::default()
+    };
+    let my_person = Person::create(&conn, &my_person_form).unwrap();
+    let local_user_form = LocalUserForm {
+      person_id: Some(my_person.id),
+      password_encrypted: Some("".to_string()),
+      ..Default::default()
+    };
+    let local_user = LocalUser::create(&conn, &local_user_form).unwrap();
+
+    // Update the users languages to all
+    LocalUserLanguage::update_user_languages(&conn, None, local_user.id).unwrap();
+
+    let post_listings_all = PostQuery::builder()
+      .conn(&conn)
+      .sort(Some(SortType::New))
+      .show_bot_accounts(Some(true))
+      .my_person_id(Some(my_person.id))
+      .build()
+      .list()
+      .unwrap();
+
+    // no language filters specified, all posts should be returned
+    assert_eq!(4, post_listings_all.len());
+
+    let french_id = Language::read_id_from_code(&conn, "fr").unwrap();
+    LocalUserLanguage::update_user_languages(&conn, Some(vec![french_id]), local_user.id).unwrap();
+    let langs = LocalUserDiscussionLanguageView::read_languages(&conn, local_user.id).unwrap();
+    dbg!(&langs);
+
+    let post_listing_french = PostQuery::builder()
+      .conn(&conn)
+      .sort(Some(SortType::New))
+      .show_bot_accounts(Some(true))
+      .my_person_id(Some(my_person.id))
+      .build()
+      .list()
+      .unwrap();
+
+    // only one french language post should be returned
+    assert_eq!(1, post_listing_french.len());
```

Comment:
This keeps failing after rebase and i cant figure out why. Already checked that the user languages are set correctly to french, and that the db queries are unchanged. It just keeps returning all 4 posts here, seemingly ignoring the user languages for no reason.

Reply:
Nevermind, forgot to pass local_user_id. Maybe later we should change it so the builder takes LocalUserView, so we dont have to pass all these things separately, and easily forget some.